### PR TITLE
Ipsec Policies for Virtual Network Gateway Connection

### PR DESCRIPTION
**Pull request by henry416**
## Description
REST spec here: https://github.com/Azure/azure-rest-api-specs/pull/956

+ Added IPsec Policies support to virtual network gateway connection model
+ Added passing tests for IPsec Policies connections
+ new api-version of Network swagger added (all changes but in 2017-03-01 version, copied over from previous rest spec version)

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by henry416** [comment]
@cormacpayne Hi, is this waiting on the spec to be merged for the CR to proceed?

**Comment by cormacpayne** [comment]
@henry416 Hey Henry, we are waiting for the spec to be merged in order to continue with the review. This ensures that all changes with the generated code are finalized, and we don't have to keep updating the review based on changes to the rest-api-specs PR

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

**Comment by henry416** [comment]
@cormacpayne OK. However can you clarify if what I have submitted test records is sufficient?

**Comment by cormacpayne** [comment]
@henry416 test coverage looks good to me 👍

**Comment by cormacpayne** [comment]
@azuresdkci test this please

**Comment by henry416** [comment]
@cormacpayne Is the current checked-in build stable?

**Comment by henry416** [comment]
@cormacpayne rest specs have been merged, this PR can proceed

File: src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Properties/AssemblyInfo.cs
Comment (cormacpayne) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Properties/AssemblyInfo.cs]: @henry416 you can set `AssemblyVersion` to 9.0.0 and keep `AssemblyFileVersion` at 9.2.0.

`AssemblyFileVersion` should always match the version found in `project.json`, and `AssemblyVersion` shares only the major version of the `AssemblyFileVersion`

File: src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/ExpressRouteCircuitPeering.cs
Comment (cormacpayne) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/ExpressRouteCircuitPeering.cs]: @henry416 I thought the position of `RouteFilter` in the constructor was fixed in a previous swagger spec PR (so that it would be the last parameter). Do you know if this change was reverted?

File: src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/VirtualNetworkGateway.cs
Comment (cormacpayne) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/VirtualNetworkGateway.cs]: @henry416 similar comment to that of `RouteFilter`. Was this position changed?

**Review by cormacpayne — CHANGES_REQUESTED** [review]
@henry416 Hey Henry, just a few comments that need clarification

File: src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/ExpressRouteCircuitPeering.cs
Comment (henry416) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/ExpressRouteCircuitPeering.cs]: I never touched the expressroute specs - I will investigate this

File: src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/VirtualNetworkGateway.cs
Comment (henry416) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/VirtualNetworkGateway.cs]: @Nilambari has a change in the parameter waiting for PR

Comment (henry416) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/VirtualNetworkGateway.cs]: @cormacpayne Confirmed that this is intended, see Nilambari's PR: https://github.com/Nilambari/azure-sdk-for-net/blob/4a1d0de7713aaf848ecc5a2f9b8b5b0aeea94d9c/src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Generated/Models/VirtualNetworkGateway.cs

**Comment by henry416** [comment]
@cormacpayne I have addressed the issues with the help of the teams that own those components + fixed the tests

File: src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Properties/AssemblyInfo.cs
Comment (henry416) [inline src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Properties/AssemblyInfo.cs]: Fixed.

**Comment by henry416** [comment]
@cormacpayne Is there any other outstanding issue with this PR?

**Comment by cormacpayne** [comment]
@henry416 did you get the email I sent? There are multiple Network PRs that have been opened in this repository and they need to be coordinated so we are not publishing multiple packages in a short time frame.

**Comment by henry416** [comment]
@cormacpayne can the merge proceed?

**Comment by henry416** [comment]
@cormacpayne Is this still in review? Once this is done we can do a double merge (my changes into branch and then branch into AutoRest master)

**Comment by jobatzil** [comment]
@henry416 Hey henry, could you maybe remove the files from your PR with no changes but a different EOF?

**Comment by henry416** [comment]
@jobatzil EOF removed and merge conflict with network-changes branch resolved

**Comment by henry416** [comment]
@cormacpayne I have resolved the merge issues, can this go through now?

**Review by cormacpayne — APPROVED** [review]

**Pull request merged**
